pulp_container/app/serializers.py

```diff
@@ -277,6 +277,22 @@ class Meta:
         model = models.ContainerRemote
 
 
+class ContainerPullThroughRemoteSerializer(RemoteSerializer):
```

Comment:
we should do something against cross type usage e.g.  container repo + remote pull-through

Reply:
You cannot combine remotes because of https://github.com/pulp/pulp_container/blob/552998fd29f89c4cf0c5a59240b22e3177ef3206/pulp_container/app/models.py#L483:
```
[lmjachky@localhost ostree-perf]$ http :5001/pulp/api/v3/repositories/container/container/ remote=/pulp/api/v3/remotes/container/pull-through/018bec87-8fdb-7ca7-bde0-7d2fad4778d0/ name=cache
HTTP/1.1 400 Bad Request
Access-Control-Expose-Headers: Correlation-ID
Allow: GET, POST, HEAD, OPTIONS
Connection: keep-alive
Content-Length: 65
Content-Type: application/json
Correlation-ID: a4ec7f0ddd8c432ca9658c380d066030
Cross-Origin-Opener-Policy: same-origin
Date: Mon, 20 Nov 2023 11:40:03 GMT
Referrer-Policy: same-origin
Server: nginx/1.22.1
Vary: Accept
X-Content-Type-Options: nosniff
X-Frame-Options: DENY

{
    "remote": [
        "Type for Remote 'cache' does not match Repository."
    ]
}
```

Reply:
yes, but try creating repo and remote separately and provide the remote during sync call

Reply:
Correct. This case was not considered. Thanks for pointing out:
```
        "description": "ContainerRemote matching query does not exist.",
        "traceback": "  File \"/src/pulpcore/pulpcore/tasking/tasks.py\", line 60, in _execute_task\n    result = func(*args, **kwargs)\n  File \"/src/pulp_container/pulp_container/app/tasks/synchronize.py\", line 36, in synchronize\n    remote = ContainerRemote.objects.get(pk=remote_pk)\n  File \"/usr/local/lib/python3.8/site-packages/django/db/models/manager.py\", line 87, in manager_method\n    return getattr(self.get_queryset(), name)(*args, **kwargs)\n  File \"/usr/local/lib/python3.8/site-packages/django/db/models/query.py\", line 637, in get\n    raise self.model.DoesNotExist(\n"
    },

```